iosApp/iosApp/Pages/StopDetails/StopDetailsPage.swift

```diff
@@ -178,19 +198,32 @@ struct StopDetailsPage: View {
         nearbyVM.setLastTripDetailsFilter(stopId, tripFilter)
     }
 
-    func updateDepartures() {
+    func updateDepartures(routeCardParams: RouteCardParams) {
         Task {
-            if stopId != stopDetailsVM.stopData?.stopId { return }
+            if routeCardParams.stopId != routeCardParams.stopData?.stopId {
+                return
+            }
             let nextRouteCardData = await stopDetailsVM.getRouteCardData(
-                stopId: stopId,
-                alerts: nearbyVM.alerts,
-                now: now,
-                isFiltered: stopFilter != nil
+                stopId: routeCardParams.stopId,
+                alerts: routeCardParams.alerts,
+                now: routeCardParams.now,
+                isFiltered: routeCardParams.stopFilter != nil
             )
             Task { @MainActor in
-                nearbyVM.setRouteCardData(stopId, nextRouteCardData)
+                nearbyVM.setRouteCardData(routeCardParams.stopId, nextRouteCardData)
                 internalRouteCardData = nextRouteCardData
             }
         }
     }
+
+    // Testing convenience
+    func updateDepartures() {
```

Comment:
I don't love this, but improving the existing tests seems hairy since they currently use `sleep`